plugin/evm/vm.go

```diff
@@ -314,6 +314,23 @@ func (vm *VM) Initialize(
 		}
 	}
 
+	vm.chainConfig = g.Config
+	vm.networkID = ethConfig.NetworkId
+
+	// Apply upgradeBytes (if any) by unmarshalling them into [chainConfig.UpgradeConfig].
+	// Initializing the chain will verify upgradeBytes are compatible with existing values.
+	if len(upgradeBytes) > 0 {
+		var upgradeConfig params.UpgradeConfig
+		if err := json.Unmarshal(upgradeBytes, &upgradeConfig); err != nil {
+			return err
+		}
+		vm.chainConfig.UpgradeConfig = upgradeConfig
+	}
+
+	// create genesisHash after applying upgradeBytes in case
+	// upgradeBytes modifies genesis.
+	vm.genesisHash = ethConfig.Genesis.ToBlock(nil).Hash()
```

Comment:
Why move this above Etherbase verifcation?

Reply:
I'm not sure if there are any side effects but feels weird to process a block before completing validation (occurs in `Genesis.ToBlock`).

Reply:
Yeah the only requirement is to include the chain config (including the upgrade bytes) in genesis.
Going to rearrange this to keep the verification. Very good idea.